components/reference-data/src/main/java/org/datacleaner/beans/ReferenceDataMatcherAnalyzerReducer.java

```diff
@@ -35,37 +30,9 @@ public BooleanAnalyzerResult reduce(Collection<? extends BooleanAnalyzerResult>
         if (partialResults.isEmpty()) {
             return null;
         }
-
-        //Create the dimensions
-        final List<CrosstabDimension> columnStatisticCrosstabDimensions = new ArrayList<CrosstabDimension>();
-        final List<CrosstabDimension> columnValueCombinationCrosstabDimensions = new ArrayList<CrosstabDimension>();
-        for (BooleanAnalyzerResult partialResult : partialResults) {
-            final Crosstab<Number> partialColumnStatisticsCrosstab = partialResult.getColumnStatisticsCrosstab();
-            final Crosstab<Number> partialValueCombinationCrosstab = partialResult.getValueCombinationCrosstab();
-            CrosstabReducerHelper.createDimensions(columnStatisticCrosstabDimensions, partialColumnStatisticsCrosstab);
-            CrosstabReducerHelper.createDimensions(columnValueCombinationCrosstabDimensions, partialValueCombinationCrosstab);
-        }
-
-        //create the crosstabs
-        final Crosstab<Number> newResultColumnStatistics = new Crosstab<Number>(Number.class,
-                columnStatisticCrosstabDimensions);
-        final Crosstab<Number> newResultColumnValueCombination = new Crosstab<Number>(Number.class,
-                columnValueCombinationCrosstabDimensions);
-
-        //add the partial results 
-        for (BooleanAnalyzerResult partialResult : partialResults) {
-            final Crosstab<Number> partialColumnStatisticsCrosstab = partialResult.getColumnStatisticsCrosstab();
-            final Crosstab<Number> partialValueCombinationCrosstab = partialResult.getValueCombinationCrosstab();
-            CrosstabReducerHelper.addData(newResultColumnStatistics, partialColumnStatisticsCrosstab);
-            CrosstabReducerHelper.addData(newResultColumnValueCombination, partialValueCombinationCrosstab);
-        }
-
-        return new BooleanAnalyzerResult(newResultColumnStatistics, newResultColumnValueCombination);
+        // the implementation of this reducer would be almost identical with
+        // BooleanAnalyzerReducer. 
+        final BooleanAnalyzerReducer booleanAnalyzerReducer = new BooleanAnalyzerReducer();
+        return booleanAnalyzerReducer.reduce(partialResults);
```

Comment:
maybe delete this class entirely and simply annotate ReferenceDataMatcherAnalyzer so that it uses the BooleanAnalyzerReducer?